source/daplink/drag-n-drop/intelhex.c

```diff
@@ -176,6 +176,9 @@ hexfile_parse_status_t parse_hex_blob(const uint8_t *hex_blob, const uint32_t he
                                             if (((next_address_to_write & 0xffff0000) | line.address) != next_address_to_write) {
                                                 load_unaligned_record = 1;
                                                 status = HEX_PARSE_UNALIGNED;
+                                                // Function will be executed again and will start by finishing to process this record by
+                                                // adding the this line into bin_buf, so the 1st loop iteration should be the next blob byte
+                                                hex_blob++;
```

Comment:
`idx--;` would also do the trick, which might be "safer" in the sense that it changes less the behaviour of this loop.

With `idx--` this part of the code would still be executed, but writing to `line.buffer[36]` (the last byte)  the same data that is there already from the previous call to `parse_hex_blob()`: https://github.com/ARMmbed/DAPLink/blob/7839c6cef09eb1b3ae66e180450753e237559bd7/source/daplink/drag-n-drop/intelhex.c#L143-L146

For that reason it felt more efficient to move `hex_blob` forward instead of `idx` backwards, but maybe I've missed anything else that might be affected by this.